src/shortestpaths/desopo-pape.jl

```diff
@@ -0,0 +1,46 @@
+"""
+    struct DEposoPapeState{T, U}
+
+An [`AbstractPathState`](@ref) designed for D`Esopo-Pape shortest-path calculations.
+"""
+struct DEsopoPapeState{T <:Real, U <: Integer} <: AbstractPathState
+    parents::Vector{U}
+    dists::Vector{T}
+end
+
+function desopo_pape_shortest_paths(g::AbstractGraph, 
+    src::Integer,
+    distmx::AbstractMatrix{T} = weights(g)) where T <: Real
+    U = eltype(g)
+    nvg = nv(g)
+    (src in 1:nvg) || throw(DomainError("src should be in between 1 and $nvg"))
```

Comment:
The constructor for `DomainError` is either `DomainError(val)` or `DomainError(val, msg)` so in your case, you probably should write `DomainError(src, "....")`.

Reply:
Ok, onto it!